cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/IOUContract.kt

```diff
@@ -30,17 +30,32 @@ class IOUContract : Contract {
      * considered valid.
      */
     override fun verify(tx: LedgerTransaction) {
-        val command = tx.commands.requireSingleCommand<Commands.Create>()
-        requireThat {
-            // Generic constraints around the IOU transaction.
-            "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
-            "Only one output state should be created." using (tx.outputs.size == 1)
-            val out = tx.outputsOfType<IOUState>().single()
-            "The lender and the borrower cannot be the same entity." using (out.lender != out.borrower)
-            "All of the participants must be signers." using (command.signers.containsAll(out.participants.map { it.owningKey }))
-
-            // IOU-specific constraints.
-            "The IOU's value must be non-negative." using (out.value > 0)
+        val command = tx.commands.requireSingleCommand<Commands>()
+
+        when (command.value) {
+            is Commands.Create -> requireThat {
+                // Generic constraints around the IOU transaction.
+                "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
+                "Only one output state should be created." using (tx.outputs.size == 1)
+                val out = tx.outputsOfType<IOUState>().single()
+                "The lender and the borrower cannot be the same entity." using (out.lender != out.borrower)
+                "All of the participants must be signers." using (command.signers.containsAll(out.participants.map { it.owningKey }))
+
+                // IOU-specific constraints.
+                "The IOU's value must be non-negative." using (out.value > 0)
+            }
+            is Commands.Destroy -> requireThat {
+                "There should be one IOU input." using (tx.inputs.size == 1)
+                "There should be no output." using (tx.outputs.isEmpty())
+
+                val inputIOU = tx.inputsOfType<IOUState>().single()
+
```

Comment:
Do you accept a negative value on the inputs?

Reply:
no, but this condition is checked on `Create`. I'm assuming that if we can create `IOUState` with only positive numbers, then on payback, when we destroying this states, there is no way to have `IOUState` with negative value. Am I wrong?

Reply:
You are not wrong. Sometimes I get paranoid, and prefer to make things explicit.

Reply:
for you it's being paranoid, for me it's "yes, I finally know something!" :D so please keep being paranoid - it really helps with studying 👍